pkg/controllers/member/serviceexport/controller.go

```diff
@@ -256,6 +264,65 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) setAzureRelatedInformation(ctx context.Context, service *corev1.Service, export *fleetnetv1alpha1.InternalServiceExport) error {
+	export.Spec.Type = service.Spec.Type
+	if service.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return nil
+	}
+	export.Spec.IsInternalLoadBalancer = service.Annotations[objectmeta.AzureLoadBalancerInternalAnnotation] == "true"
+	if export.Spec.IsInternalLoadBalancer {
+		// no need to populate the PublicIPResourceID and IsDNSLabelConfigured which are only applicable for external load balancer
+		return nil
+	}
+
+	serviceKObj := klog.KObj(service)
+	if len(service.Status.LoadBalancer.Ingress) == 0 {
+		// Assuming once the service status is updated, the controller will be triggered again.
+		klog.V(2).InfoS("The load balancer IP is not assigned yet", "service", serviceKObj)
+		return nil
+	}
+
+	if service.Status.LoadBalancer.Ingress[0].IP == "" {
+		err := errors.New("the service ingress is not nil but with empty IP")
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(err), "Failed to get the load balancer IP from service", "service", serviceKObj, "status", service.Status)
+		return nil
```

Comment:
Do we want to return err here?

Reply:
Assume it's not possible in production, so logged as unexpected error.

If the IP of service is updated, the controller will be retriggered.